crates/shared/src/trade_finding/external.rs

```diff
@@ -93,15 +93,21 @@ impl ExternalTradeFinder {
                     .text()
                     .await
                     .map_err(|err| PriceEstimationError::EstimatorInternal(anyhow!(err)))?;
-                serde_json::from_str::<dto::Quote>(&text)
-                    .map(Trade::from)
-                    .map_err(|err| {
-                        if let Ok(err) = serde_json::from_str::<dto::Error>(&text) {
-                            PriceEstimationError::from(err)
-                        } else {
-                            PriceEstimationError::EstimatorInternal(anyhow!(err))
-                        }
-                    })
+                let quote = serde_json::from_str::<dto::Quote>(&text).map_err(|err| {
```

Comment:
I don't think the deserializer should make any business logic decisions.

Reply:
I think all this custom deserialization logic makes this too awkward. Why not parse the responses into an enum as advertised by the openapi spec and have the external trade finder reject responses with the new format? That way you only have to implement a `todo!()` instead of replacing a bunch of deserialization logic.

Reply:
Tried to address it.

Reply:
After removing the custom deserializer, I remembered that won't work for the current solver responses. So I think we still need it.